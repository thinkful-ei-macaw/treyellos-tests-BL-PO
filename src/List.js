import React from 'react';
import Card from './card.js';

function List(props) {
    const cards = props.cards.map(card =>
        <Card key={card.id} title={card.title} content={card.content}/>
        );
    return (
        <section className="List">
            <header className="List-header">
                <h2>{props.header}</h2>
            </header>
            <div className="List-cards">
                {cards}
                <button type="button" class="list-add-card">+Add a Card</button>
            </div>
        </section>
        
    );    
}

// use default props for tests
List.defaultProps = {
    cards: []
}

export default List;