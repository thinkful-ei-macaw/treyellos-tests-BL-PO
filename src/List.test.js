import React from 'react';
import ReactDOM, { unstable_renderSubtreeIntoContainer } from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';

describe('DOM renders all elements', () => {
    it('DOM renders all elements', () => {
        const div = document.createElement('div');
        ReactDOM.render(<List />, div);
        ReactDOM.unmountComponentAtNode(div);
    })
    it('Renders without crashes', () => {
        const tree = renderer
        .create(<List headers='my-list' cards={[
            {
                id: 1,
                title: 'myCard',
                content: 'myContent'
            }
        ]} />)
        expect(tree).toMatchSnapshot();
    })
});

