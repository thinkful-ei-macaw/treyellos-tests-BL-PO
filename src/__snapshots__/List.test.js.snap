// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOM renders all elements Renders without crashes 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2 />
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Cards"
    >
      <button
        type="button"
      >
        Delete
      </button>
      <h3>
        myCard
      </h3>
      <p>
        myContent
      </p>
    </div>
    <button
      class="list-add-card"
      type="button"
    >
      +Add a Card
    </button>
  </div>
</section>
`;
